operator/data/profiles/demo.yaml

```diff
@@ -137,4 +137,40 @@ spec:
           targetPort: 15443
           name: tls
     kiali:
+      enabled: true
```

Comment:
For the enablement setting and k8s setting defined here: https://github.com/istio/api/blob/master/operator/v1alpha1/component.proto#L106. It would be better to move to the addonComponents spec above: https://github.com/istio/istio/pull/20131/files#diff-2e09cf6abd93e7c2183f516ed0c7f948R64

Reply:
@richardwxn I just want to confirm what you are asking me to change.

Do you mean that these settings:

```
kiali.operator.enabled
kiali.operator.nodeSelector
kiali.operator.podAnnotations
kiali.operator.resources
kiali.operator.tolerations
kiali.operator.podAntiAffinityLabelSelector
kiali.operator.podAntiAffinityTermLabelSelector

kiali.enabled
kiali.nodeSelector
kiali.podAnnotations
kiali.resources
kiali.tolerations
kiali.podAntiAffinityLabelSelector
kiali.podAntiAffinityTermLabelSelector
kiali.replicaCount
```

Should be moved from where they are now in `operator/data/profiles/demo.yaml` up into the "addonComponents.kiali" section? So it would look like:

```
addonComponents:
  kiali:
    operator:
      enabled: true
      nodeSelector: {}
      podAnnotations: {}
      resources: {}
      tolerations: {}
      podAntiAffinityLabelSelector: []
      podAntiAffinityTermLabelSelector: []
    enabled: true
    nodeSelector: {}
    podAnnotations: {}
    resources: {}
    tolerations: {}
    podAntiAffinityLabelSelector: []
    podAntiAffinityTermLabelSelector: []
    replicaCount: 1
...
values:
  kiali:
    operator:
      ...the kiali.operator settings NOT found above in addonComponents.kiali.operator ...
    createDemoSecret: true    
    ...and the rest of the kiali settings NOT found above in addonComponents.kiali ...
```

Reply:
BTW: I don't see any other addonComponents using these "k8s" settings in manifests/, so I'm not sure if this stuff is wired in fully or not. If I do a grep looking for, say, podAnnotations, I don't see any using it:

```
$ grep -R "\.k8s\.podAnnotations" manifests/*
```

But if I look at individual components, they are using their own versions of these settings:

```
$ grep -R "\.podAnnotations" manifests/*
...
manifests/istio-telemetry/prometheus/templates/deployment.yaml:        {{- if .Values.prometheus.podAnnotations }}
manifests/istio-telemetry/prometheus/templates/deployment.yaml:{{ toYaml .Values.prometheus.podAnnotations | indent 8 }}
...
manifests/istio-telemetry/tracing/templates/deployment-jaeger.yaml:        {{- if .Values.tracing.jaeger.podAnnotations }}
manifests/istio-telemetry/tracing/templates/deployment-jaeger.yaml:{{ toYaml .Values.tracing.jaeger.podAnnotations | indent 8 }}
...
manifests/istio-telemetry/grafana/templates/deployment.yaml:        {{- if .Values.grafana.podAnnotations }}
manifests/istio-telemetry/grafana/templates/deployment.yaml:{{ toYaml .Values.grafana.podAnnotations | indent 8 }}
```

And if I look at the proto, I do see these defined in their own config (not relying on, say, some common component) - for example here in [PrometheusConfig](https://github.com/istio/istio/blob/cda6b101788593bb91e920cc36e12daa914f6c35/operator/pkg/apis/istio/v1alpha1/values_types.proto#L1403) and here in [TracingJaegerConfig](https://github.com/istio/istio/blob/cda6b101788593bb91e920cc36e12daa914f6c35/operator/pkg/apis/istio/v1alpha1/values_types.proto#L2025).

So, moving these to addonComponents into an k8s block seems to be something that isn't done for any other of the third-party addon components (at least, I don't see Grafana or Jaeger or Prometheus using these addonComponent `k8s` settings - they all use their own within `.Values.<component>` as you see above in the grep results - you can see things like `.Values.prometheus.podAnnotations` where I would expect to see `.Values.prometheus.k8s.podAnnotations`.

Am I misunderstanding how this is working? Is there some magic that is converting these addonComponent k8s settings into `.Values.<component>` settings (without the intermediate `k8s` level)?

Reply:
@richardwxn can you answer this question please.